### PR TITLE
[PUBDEV-7421] ability to prepare hive token before h2o submit

### PR DESCRIPTION
**Pull request by honzasterba**
feature aimed mainly at steam and sw

### CODE REVIEW

File: scripts/jenkins/groovy/kerberosCommands.groovy [hunk 1]
```diff
@@ -39,15 +45,26 @@ private GString getCommandHadoop(final stageConfig, final spnegoAuth, final bool
...
     def securityArgs = ""
-    if (steam) {
+    if (impersonate) {
         securityArgs = "-principal steam/localhost@H2O.AI -keytab /etc/hadoop/conf/steam.keytab -run_as_user jenkins"
     }
+    def tokenPreparation = ""
+    def usePreparedToken = ""
+    if (prepareToken) {
+        tokenPreparation = """export HADOOP_CLASSPATH=\$(cat /opt/hive-jdbc-cp)
+            hadoop jar h2o-hadoop-3/h2o-cdh6.3-assembly/build/libs/h2odriver.jar
```
Comment (michalkurka) [inline scripts/jenkins/groovy/kerberosCommands.groovy:55]: Should the path to the jar be parametrized the same way as it is on https://github.com/h2oai/h2o-3/pull/4479/files#diff-6e5d07024fa7974b3ccb6899740e102bL48?

Comment (honzasterba) [reply scripts/jenkins/groovy/kerberosCommands.groovy:55]: good catch!

File: h2o-hive/src/main/java/water/hive/DelegationTokenRefresher.java [hunk 2]
```diff
@@ -28,11 +28,18 @@
...
     }
+    String token = conf.get(H2O_HIVE_TOKEN);
+    if (token != null) {
+      log("Adding credentials from property", null);
+      Credentials creds = HiveTokenGenerator.tokenToCredentials(token);
+      UserGroupInformation.getCurrentUser().addCredentials(creds);
+    }
     String authUser = conf.get(H2O_AUTH_USER);
     String authPrincipal = conf.get(H2O_AUTH_PRINCIPAL);
     String authKeytab = conf.get(H2O_AUTH_KEYTAB);
```
Comment (mn-mikke) [inline h2o-hive/src/main/java/water/hive/DelegationTokenRefresher.java:45]: Does `H2O_AUTH_KEYTAB` still need to be passed into setup method?

Comment (honzasterba) [reply h2o-hive/src/main/java/water/hive/DelegationTokenRefresher.java:45]: no , you can use use H2O_HIVE_TOKEN H2O_HIVE_JDBC_URL_PATTERN H2O_HIVE_PRINCIPAL H2O_HIVE_HOST

**Pull request merged**